### ISSUE TITLE
`match_wildcard_for_single_variants` does not consider `#[non_exhaustive]` enums

### ISSUE THREAD

**Issue by WiSaGaN** [rust-lang/rust-clippy#14249]
### Problem Description

The [`match_wildcard_for_single_variants`](https://rust-lang.github.io/rust-clippy/master/index.html#/match_wildcard_for_single_variants) lint currently ignores `#[non_exhaustive]` enums. This allows matches to use wildcard patterns (`_ => ...`) even when all existing variants could be explicitly matched, which contradicts the intent of developers who want exhaustive handling of *current* variants while still allowing future additions.

### Proposed Solution

Adjust the lint to:
1. Check if the enum is marked `#[non_exhaustive]`
2. Warn when a wildcard is used *and* all current variants are explicitly matched
3. Add a configuration option (e.g., `enforce-exhaustive-non-exhaustive`) to control this behavior

### Lint Name

match_wildcard_for_single_variants

### Reproducer

I tried this code:

```rust
#[non_exhaustive]
enum MyEnum { A, B }

fn main() {
    let x = MyEnum::A;
    
    // Should warn about wildcard usage (but currently doesn't)
    match x {
        MyEnum::A => println!("A"),
        _ => println!("Wildcard"), // ❌ No warning despite only 2 variants
    }
}
```

I expected to see this happen:
Warns about Enum::B not handled when running clippy

Instead, this happened:
No warning

### Version

```text
rustc 1.84.1 (e71f9a9a9 2025-01-27)
binary: rustc
commit-hash: e71f9a9a98b0faf423844bf0ba7438f29dc27d58
commit-date: 2025-01-27
host: x86_64-unknown-linux-gnu
release: 1.84.1
LLVM version: 19.1.5
```

**Comment by samueltardieu** [comment]
Wouldn't this use case be covered by `wildcard_enum_match_arm` lint already?

**Comment by WiSaGaN** [comment]
> Wouldn't this use case be covered by `wildcard_enum_match_arm` lint already?

I didn't know this lint. Thanks! I will close it shortly.
